packages/replayio/src/utils/installation/installLatestRelease.ts

```diff
@@ -15,11 +17,33 @@ export async function installLatestRelease() {
   const runtimePath = getReplayPath("runtimes", runtimeMetadata.destinationName);
   const downloadFilePath = getReplayPath("runtimes", runtimeMetadata.downloadFileName);
 
+  const metadata = {
+    attemptNumber: 0,
+  };
+
+  const downloadPromise = downloadReplayFile(5, metadata);
+
+  logPromise(downloadPromise, {
+    messages: {
+      failed: "Something went wrong installing the Replay browser. Please try again later.",
+      pending: () => {
+        const prefix = `Downloading from ${link("static.replay.io")}`;
+        let suffix = "";
+        if (metadata.attemptNumber > 1) {
+          suffix = `${dim(` (retry ${metadata.attemptNumber - 1} of 4)`)}`;
+        }
+
+        return `${prefix}${suffix}...`;
+      },
+      success: "Replay browser has been updated.",
+    },
+  });
+
+  const buffers = await downloadPromise;
+
   debug("Removing previous installation at %s", runtimePath);
   rmSync(runtimePath, { force: true, recursive: true });
```

Comment:
We shouldn't remove the old installation until we've successfully downloaded the new one, else a failed update will make it so you can't even use Replay anymore.